job-server/src/spark.jobserver/WebApi.scala

```diff
@@ -273,6 +273,33 @@ class WebApi(system: ActorSystem,
               }
             }
           }
+        } ~
+        put {
+          parameter("reset") { reset =>
+            respondWithMediaType(MediaTypes.`text/plain`) { ctx =>
+              reset match {
+                case "all" => {
+                  import ContextSupervisor._
+                  import collection.JavaConverters._
+                  logger.warn("refreshing contexts")
+                  val future = (supervisor ? ListContexts).mapTo[Seq[String]]
+                  Await.result(future, 1 second).asInstanceOf[Seq[String]]
+                    .foreach { context => {
+                      val stopFuture = supervisor ? StopContext(context)
+                      Await.ready(stopFuture, 5 seconds)
+                    }
+                  }
+                  Thread.sleep(1000) // we apparently need some sleeping in here, so spark can catch up
+
+                  (supervisor ? AddContextsFromConfig).onFailure {
```

Comment:
Hmm.  What if you had more contexts open than the ones listed in the config?  Say in config you have “MyContext”, “YourContext” and now I add “EvansContext”.  

I think `reset=config` and only resetting the contexts from the configs would be the most consistent behavior.  What do others think?

> On Oct 15, 2015, at 2:34 PM, David <notifications@github.com> wrote:
> 
> In job-server/src/spark.jobserver/WebApi.scala <https://github.com/spark-jobserver/spark-jobserver/pull/284#discussion_r42183627>:
> 
> > +            respondWithMediaType(MediaTypes.`text/plain`) { ctx =>
> > +              reset match {
> > +                case "all" => {
> > +                  import ContextSupervisor._
> > +                  import collection.JavaConverters._
> > +                  logger.warn("refreshing contexts")
> > +                  val future = (supervisor ? ListContexts).mapTo[Seq[String]]
> > +                  Await.result(future, 1 second).asInstanceOf[Seq[String]]
> > +                    .foreach { context => {
> > +                      val stopFuture = supervisor ? StopContext(context)
> > +                      Await.ready(stopFuture, 5 seconds)
> > +                    }
> > +                  }
> > +                  Thread.sleep(1000) // we apparently need some sleeping in here, so spark can catch up
> > +
> > +                  (supervisor ? AddContextsFromConfig).onFailure {
> Correct, this is the desired behavior - the config has the right sizing for our larger jobs, so re-loading from that makes the most sense. As you mention, 'ListContexts' doesn't have enough detail in it to re-create the contexts at their preferred sizes.
> 
> Would it make sense to go with a different param value for this, like "reset=config" instead of "all"?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/spark-jobserver/spark-jobserver/pull/284/files#r42183627>.
>